src/creatures/players/imbuements/imbuements.cpp

```diff
@@ -344,17 +356,50 @@ Category* Imbuements::getCategoryByID(uint16_t id)
 	return it != categories.end() ? &*it : nullptr;
 }
 
-std::vector<Imbuement*> Imbuements::getImbuements(Player* player, Item* item)
+bool Imbuements::parseImbuements(Player* player, Item* item)
 {
-	std::vector<Imbuement*> filtered;
-	for (auto& info : imbues) {
+	if (!player || !item) {
+		return false;
+	}
+
+	for (auto& info : imbues)
+	{
 		Imbuement* imbuement = &info.second;
-		if (!g_events->eventPlayerCanBeAppliedImbuement(player, imbuement, item)) {
-			continue;
+
+		const ItemType& it = Item::items[item->getID()];
+		uint8_t slot = it.imbuingSlots;
+		if (slot <= 0)
```

Comment:
This validation does nothing, your for already make you safe, since slot id will never be < than slot. Also slot is uint8, so slot is never < 0. So you can safely remove this if.